lib/ansible/modules/system/listen_ports_facts.py

```diff
@@ -0,0 +1,264 @@
+        if listening_search:
+            splitted = line.split()
+            conns = re.search('([^ ]+):([0-9]+)', splitted[3])
+            pidstr = ''
+            if 'tcp' in splitted[0]:
+                protocol = 'tcp'
+                pidstr = splitted[6]
+            elif 'udp' in splitted[0]:
+                protocol = 'udp'
+                pidstr = splitted[5]
+            pids = re.search('(([0-9]+)/(.*)|-)', pidstr)
+            if conns and pids:
+                address = conns.group(1)
+                port = conns.group(2)
+                if (pids.group(2)):
+                    pid = pids.group(2)
+                else:
+                    pid = 0
+                if (pids.group(3)):
+                    name = pids.group(3)
+                else:
+                    name = ''
+                result = dict(pid=int(pid), address=address, port=int(port), protocol=protocol, name=name)
+                if result not in results:
+                    results.append(result)
+            else:
+                raise EnvironmentError('Could not get process information for the listening ports.')
+    return results
+
+
+def applyWhitelist(portspids, whitelist=None):
+    if whitelist is None:
+        whitelist = list()
+    processes = list()
+    for p in portspids:
+        if int(p['port']) not in whitelist and str(p['port']) not in whitelist:
+            process = dict(pid=p['pid'], address=p['address'], port=p['port'], protocol=p['protocol'], name=p['name'], stime=p['stime'], user=p['user'])
+            if process not in processes:
+                processes.append(process)
+    return processes
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            whitelist_tcp=dict(type='list', default=list()),
+            whitelist_udp=dict(type='list', default=list()),
+        ),
+        supports_check_mode=True,
+    )
+
+    if platform.system() != 'Linux':
+        module.fail_json(msg='This module requires Linux.')
+
+    def getPidSTime(pid):
+        ps_cmd = module.get_bin_path('ps', True)
+        rc, ps_output, stderr = module.run_command([ps_cmd, '-o', 'lstart', '-p', str(pid)])
+        stime = ''
+        if rc == 0:
+            for line in iter(ps_output.splitlines()):
+                if 'started' not in line:
+                    stime = line
+        return stime
+
+    def getPidUser(pid):
+        ps_cmd = module.get_bin_path('ps', True)
+        rc, ps_output, stderr = module.run_command([ps_cmd, '-o', 'user', '-p', str(pid)])
+        user = ''
+        if rc == 0:
+            for line in iter(ps_output.splitlines()):
+                if line != 'USER':
+                    user = line
+        return user
+
+    result = dict(
+        changed=False,
+        ansible_facts=dict(
+            tcp_listen_violations=list(),
+            udp_listen_violations=list(),
+            tcp_listen=list(),
+            udp_listen=list(),
+        )
```

Comment:
Not mandatory, but a good practice: a trailing comma here too ;-)
```suggestion
        ),
```